Product/Production/Services/PatientDiscoveryCore/src/test/java/gov/hhs/fha/nhinc/patientdiscovery/parser/TestPatientDiscoveryMessageHelper.java

```diff
@@ -0,0 +1,345 @@
+import org.hl7.v3.PRPAMT201306UV02QueryByParameter;
+import org.hl7.v3.PRPAMT201310UV02Patient;
+import org.hl7.v3.ParticipationTargetSubject;
+import org.hl7.v3.XActMoodIntentEvent;
+
+/**
+ * The class is used to create PD request and response messages to facilitate testing.
+ *
+ * @author tjafri
+ */
+public class TestPatientDiscoveryMessageHelper {
+
+    public static final String ROOT = "1.1";
+    public static final String EXTENSION = "D123401";
+    public static final String QUERY_ID_ROOT = "2.2";
+    public static final String QUERY_ID_EXTENSION = "abd3453dcd24wkkks545";
+
+    public static PRPAIN201305UV02 createPRPAIN201305UV02Request() {
+        PRPAIN201305UV02 request = new PRPAIN201305UV02();
+        request.setControlActProcess(createControlActProcess());
+        return request;
+    }
+
+    public static PRPAIN201306UV02 createPRPAIN201306UV02Response() {
+        PRPAIN201306UV02 response = new PRPAIN201306UV02();
+        response.setControlActProcess(createResponseControlActProcess());
+        return response;
+    }
+
+    private static PRPAIN201305UV02QUQIMT021001UV01ControlActProcess createControlActProcess() {
+        II subjectId = new II();
+        subjectId.setRoot(ROOT);
+        subjectId.setExtension(EXTENSION);
+
+        CD code = new CD();
+        code.setCode("PRPA_TE201306UV02");
+        code.setCodeSystem("2.16.840.1.113883.1.6");
+
+        PRPAIN201305UV02QUQIMT021001UV01ControlActProcess controlActProcess
+            = new PRPAIN201305UV02QUQIMT021001UV01ControlActProcess();
+        controlActProcess.setClassCode(ActClassControlAct.CACT);
+        controlActProcess.setMoodCode(XActMoodIntentEvent.EVN);
+        controlActProcess.setCode(code);
+        controlActProcess.setQueryByParameter(createQueryParams("Allen", "Wanderson", "M", "09-8-1971", subjectId));
+        return controlActProcess;
+    }
+
+    private static JAXBElement<PRPAMT201306UV02QueryByParameter> createQueryParams(String firstName, String lastName,
+        String gender, String birthTime, II subjectId) {
+
+        II id = new II();
+        id.setRoot(QUERY_ID_ROOT);
+        id.setExtension(QUERY_ID_EXTENSION);
+        CS statusCode = new CS();
+        statusCode.setCode("new");
+
+        PRPAMT201306UV02QueryByParameter params = new PRPAMT201306UV02QueryByParameter();
+        params.setQueryId(id);
+        params.setStatusCode(statusCode);
+        params.setParameterList(createParamList(firstName, lastName, gender, birthTime, subjectId));
+
+        return new JAXBElement<>(new QName("urn:hl7-org:v3", "queryByParameter"),
+            PRPAMT201306UV02QueryByParameter.class, params);
+    }
+
+    private static PRPAMT201306UV02ParameterList createParamList(String firstName, String lastName, String gender,
+        String birthTime, II subjectId) {
+
+        PRPAMT201306UV02ParameterList paramList = new PRPAMT201306UV02ParameterList();
+
+        // Set the Subject Gender Code
```

Comment:
I know I'm usually the one calling for more comments in the code, but -- the comments in this function in particular are unnecessary, the methods are self-documenting by name.